Producer-Consumer problem/src/org/webonise/producerconsumerproblem/Producer.java

```diff
@@ -0,0 +1,55 @@
+package org.webonise.producerconsumerproblem;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+public class Producer implements Runnable{
+    private final ArrayList<Integer> sharedItem;
+    private int size;
+    public Producer(ArrayList<Integer>sharedItem,int size){
+        // sets values of fields.
+        this.sharedItem=sharedItem;
+        this.size=size;
+    }
+    @Override
+    public void run() {
+        for(int i=0;i<7;i++){
+            System.out.println("Produced:"+i);
+            try {
+                // produces the Product.
+                produce(i);
+                display();
+            }catch(Exception e){
+                e.printStackTrace();
+            }
+        }
+    }
+    private void produce(int productNumber) throws Exception{
+        // Wait if size of sharedItem is full.
+        while(sharedItem.size()==size){
```

Comment:
Why `while` loop is used here? Don't you want to just verify if sharedItem is full with `if` statement?